eyeshade/workers/snapshots.js

```diff
@@ -0,0 +1,147 @@
+const uuidV5 = require('uuid/v5')
+const cron = require('cron-parser')
+const _ = require('underscore')
+const { BigNumber, justDate } = require('bat-utils/lib/extras-utils')
+
+exports.initialize = async (debug, runtime) => {
+  await runtime.queue.create('update-snapshot-accounts')
+}
+
+// select * from account_balances
+const getAllAccountBalancesBefore = `
+select
+  account_transactions.account_type as "accountType",
+  account_transactions.account_id as "accountId",
+  coalesce(sum(account_transactions.amount), 0.0) as "balance"
+from account_transactions
+  where created_at < $1
+group by (account_transactions.account_type, account_transactions.account_id);
+`
+
+const getAllAccountBalances = `
+select * from account_balances
+`
+
+const upsertBalanceSnapshotAccounts = `
+insert into balance_snapshot_accounts (id, created_at, snapshot_id, account_id, account_type, balance)
+values ($1, $2, $3, $4, $5, $6)
+on conflict (id)
+do update
+  set balance = $6
+returning *;
+`
+
+const updateBalanceSnapshotWithTotals = `
+update balance_snapshots
+  set
+    completed = $2,
+    updated_at = $3
+where id = $1
+returning *;
+`
+
+exports.createDailySnapshot = createDailySnapshot
+exports.updateSnapshotAccounts = updateSnapshotAccounts
+
+exports.initialize = async (debug, runtime) => {
+  if ((typeof process.env.DYNO !== 'undefined') && (process.env.DYNO !== 'worker.1')) return
+
+  daily(debug, runtime)
+}
+
+function daily (debug, runtime) {
+  const interval = cron.parseExpression('* 0 * * *', {
+    utc: true
+  })
+  const date = interval.next().getTime()
+  setTimeout(() => createDailySnapshot(debug, runtime, date, daily), date - _.now())
+}
+
+async function createDailySnapshot (debug, runtime, date, next) {
+  const until = justDate(new Date(date))
+  const snapshotId = uuidV5(until, 'dc0befa2-37a4-4235-a5ce-dfc7d5408a78').toLowerCase()
+  const result = await updateSnapshotAccounts(debug, runtime, {
+    snapshotId,
+    until
+  })
+  next && next(debug, runtime)
+  return result
+}
```

Comment:
So if we have Publishers driving the generation of the snapshots each month, I don't think I can see a use case for regularly generating snapshots.  I think we can remove the cron code for now